07_safety/01_agency_geo_county.Rmd

```diff
@@ -0,0 +1,731 @@
+                ~ ifelse(. == "-6", NA, .)))
+
+```
+
+### 2.2 Convert NIBRS state codes to state FIPS codes
+The state identifiers included in NIBRS are state abbreviation and a state code. NIBRS state codes are in alphabetical order and do not follow FIPS codes. We manually create a crosswalk between state FIPS codes and NIBRS codes by looking at the codebook (`ICPSR_38566\DS0001\38566-0001-Codebook`) and then join state FIPS codes onto the Batch Header File.
+```{r, message=FALSE}
+# Create "crosswalk" for NIBRS state codes to state FIPS codes.
+nibrs_states <- tribble(
+  ~state, ~state_abb, ~state_num_nibrs, ~state_name,
+  "01",	"AL", "01", "Alabama",
+  "02",	"AK", "50", "Alaska",
+  "04",	"AZ", "02", "Arizona", 
+  "05",	"AR", "03", "Arkansas",	
+  "06",	"CA", "04", "California",	
+  "08",	"CO", "05", "Colorado",
+  "09",	"CT", "06", "Connecticut",
+  "10",	"DE", "07", "Delaware",
+  "11",	"DC", "08", "District of Columbia",
+  "12",	"FL", "09", "Florida",
+  "13",	"GA", "10", "Georgia",	
+  "15",	"HI", "51", "Hawaii",
+  "16",	"ID", "11", "Idaho",
+  "17",	"IL", "12", "Illinois",
+  "18",	"IN", "13", "Indiana",
+  "19",	"IA", "14", "Iowa",	
+  "20",	"KS", "15", "Kansas",	
+  "21",	"KY", "16", "Kentucky",	
+  "22",	"LA", "17", "Louisiana",
+  "23",	"ME", "18", "Maine",
+  "24",	"MD", "19", "Maryland",
+  "25",	"MA", "20", "Massachusetts",
+  "26",	"MI", "21", "Michigan",	
+  "27",	"MN", "22", "Minnesota",	
+  "28",	"MS", "23", "Mississippi",	
+  "29",	"MO", "24", "Missouri",	
+  "30",	"MT", "25", "Montana",	
+  "31",	"NE", "26", "Nebraska",	
+  "32",	"NV", "27", "Nevada",	
+  "33",	"NH", "28", "New Hampshire",	
+  "34",	"NJ", "29", "New Jersey",	
+  "35",	"NM", "30", "New Mexico",	
+  "36",	"NY", "31", "New York",	
+  "37",	"NC", "32", "North Carolina",
+  "38",	"ND", "33", "North Dakota",	
+  "39",	"OH", "34", "Ohio",	
+  "40",	"OK", "35", "Oklahoma",	
+  "41",	"OR", "36", "Oregon",	
+  "42",	"PA", "37", "Pennsylvania",	
+  "44",	"RI", "38", "Rhode Island",	
+  "45",	"SC", "39", "South Carolina",	
+  "46",	"SD", "40", "South Dakota",
+  "47",	"TN", "41", "Tennessee",
+  "48",	"TX", "42", "Texas",	
+  "49",	"UT", "43", "Utah",	
+  "50",	"VT", "44", "Vermont",	
+  "51",	"VA", "45", "Virginia",	
+  "53",	"WA", "46", "Washington",	
+  "54",	"WV", "47", "West Virginia",	
+  "55",	"WI", "48", "Wisconsin",	
+  "56",	"WY", "49", "Wyoming"
+)
+
+# The NIBRS data abbreviates Nebraska as "NB" instead of its correct abbreviation "NE"
+batch_header_file <- batch_header_file %>%
+  mutate(state_abb = if_else(state_abb=="NB", "NE", state_abb))
+
+# Join state FIPS codes and full state name onto the NIBRS data
+batch_header_file <- batch_header_file %>% 
+  left_join(
+    y = nibrs_states, by = c("state_abb", "state_num_nibrs")
+  )
+
+```
+
+
+There are 51 unique combinations of `state`/`state_name`/`state_abb`/`state_num_nibrs`, so the manual crosswalk does not seem to have any errors.
+```{r, message=FALSE}
```

Comment:
what you have works, but you could also turn it into a test like: 
`stopifnot(
  batch_header_file %>%
    group_by(state, state_name, state_abb, state_num_nibrs) %>%
    summarize() %>% 
    nrow() == 51)`

Reply:
Thanks for the tip! Implemented this change.